### PR TITLE
[COOK-2730] Allow specifying the proxy_pass domain manually incase the app name doesn't match

### PR DESCRIPTION
**Pull request by benlangfeld**
http://tickets.opscode.com/browse/COOK-2730

### CODE REVIEW

File: templates/default/load_balancer.conf.erb [hunk 1]
```diff
@@ -18,6 +18,9 @@ server {
   }
   <% end %>
   location / {
-    proxy_pass http://<%= @resource.application.name %>;
+    proxy_pass http://<%= @resource.proxy_pass || @resource.application.name %>;
+    <% if @resource.set_host_header %>
+    proxy_set_header Host $host;
```
Comment (josephholsten) [inline templates/default/load_balancer.conf.erb]: when we use `proxy_set_header`, we also have the line:

```
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
```

**Comment by josephholsten** [comment]
I've created a ticket for you: http://tickets.opscode.com/browse/COOK-2730

**Comment by sethvargo** [comment]
@benlangfeld there are some comments on the JIRA ticket that require your attention.

**Comment by benlangfeld** [comment]
I've added some documentation for set_host_header, and reverted the proxy_pass change since it wasn't actually the appropriate solution to my problem.

Feel free to squash to a single commit. Anything else required?

**Comment by sethvargo** [comment]
@benlangfeld this does not merge cleanly. Could you please rebase and mark the ticket as Fix Provided?

**Comment by benlangfeld** [comment]
Rebased and squashed to a single commit.

**Comment by someara** [comment]
merged at 7b32e59e8c6092d9d35f854a188e4367a90537d9

**Pull request closed without merging**